lib/Backends/BackendUtils.cpp

```diff
@@ -81,6 +74,38 @@ runtime::RuntimeBundle::getSymbolInfo(const Named *v) const {
   return it->second;
 }
 
+/// If \p PH is an output placeholder, \returns true.
+/// This is determined by checking if the PH has a user which uses the PH as an
+/// overwritten input.
+bool checkIfOutput(const Placeholder *PH) {
+  for (auto &use : PH->getUsers()) {
+    // Look through the inputs of the PH's users. If an input is overwritten
+    // check if it's the PH, if it is return true.
+    for (unsigned i = 0; i < use.getUser()->getNumInputs(); i++) {
+      if (use.getUser()->isOverwrittenNthInput(i)) {
+        auto input = use.getUser()->getNthInput(i);
+        if (auto overwritten = llvm::dyn_cast<Placeholder>(input)) {
```

Comment:
I think you can just check `input == PH`, there’s no need to cast to do a pointer equality check.